src/geometry.nim

```diff
@@ -496,21 +465,74 @@ proc newRotZ*(angle: float32): Rotation =
     ]
 
 
+proc apply*[T](transf: Transformation, x: T): T =
+    case transf.kind
+    of tkIdentity: return x
+
+    of tkGeneric, tkComposition, tkRotation: 
```

Comment:
I don't get why we tkGeneric and tkComposition are different kinds: if you compose different kinds of transformations what you get as a result is a generic one, no? I would eliminate tkComposition kind.

Reply:
`tkGeneric` is used when you have the matrices and you don't know what type of transformation are involved.  `tkComposition` is used when you have two transformation and you generate a new generic transformation but at least you know is a transformation.

The difference should be remarked by rewriting the `Transformation` as follows:
```nim

type 
    TransformationKind* = enum
        tkIdentity, tkGeneric, tkTranslation, tkScaling, tkRotation, tkComposition

    Transformation* = object
        case kind*: TransformationKind
        of tkIdentity: discard
        of tkGeneric, tkTranslation, tkScaling, tkRotation:
            mat*, inv_mat*: Mat4f
        of tkComposition: 
            transformations: seq[Transformation]
``` 
This approch also let us do no memory allocation for `tkIdentity` and to have access to the different transformation that forms a `tkComposition`.

Reply:
That's a good way to keep composition kind alive